app/uk/gov/hmrc/fileupload/read/envelope/model.scala

```diff
@@ -66,6 +73,18 @@ object Envelope {
     val rawData = json.asInstanceOf[JsObject] ++ Json.obj("_id" -> _id)
     Json.fromJson[Envelope](rawData).get
   }
+
```

Comment:
I agree on the renaming of load constraints. Is all those gymnastics in the method to do the load necessary ? (couldn't you do this all with named and default params ? so setup the Contraints type with the default values and only populate the Some ones in the constructor ?)